From 62addaf1ef7d7fc537bc5a5a9d25eb5ce0e1d844 Mon Sep 17 00:00:00 2001
From: Mark Ryan <mark.d.ryan@intel.com>
Date: Thu, 22 Jan 2015 18:30:12 +0100
Subject: [PATCH] ov2680/ov5648: Enabled the set_exposure functions

The functions were previously disabled as the HAL was sending
incorrect exposure values to the driver which resulted in an
almost entirely black image.  It turns out the HAL was doing
this as the drivers were providing the HAL with an incorrect
exposure range.   This patch fixes the exposure range and
re-enables the set_exposure functions.

Change-Id: Iec4f6c87e78ee3f2e4688601ec8f2c81eba5c1f0
Signed-off-by: Mark Ryan <mark.d.ryan@intel.com>
---
 drivers/media/i2c/atomisp/ov2680.c | 2 +-
 drivers/media/i2c/atomisp/ov2680.h | 2 +-
 drivers/media/i2c/atomisp/ov5648.c | 2 +-
 drivers/media/i2c/atomisp/ov5648.h | 2 +-
 4 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/drivers/media/i2c/atomisp/ov2680.c b/drivers/media/i2c/atomisp/ov2680.c
index 910b31a..aec55c2 100644
--- a/drivers/media/i2c/atomisp/ov2680.c
+++ b/drivers/media/i2c/atomisp/ov2680.c
@@ -533,7 +533,7 @@ static long ov2680_s_exposure(struct v4l2_subdev *sd,
 	}
 
 	// EXPOSURE CONTROL DISABLED FOR INITIAL CHECKIN, TUNING DOESN'T WORK
-	return 0; // ov2680_set_exposure(sd, coarse_itg, analog_gain, digital_gain);
+	return ov2680_set_exposure(sd, coarse_itg, analog_gain, digital_gain);
 }
 
 
diff --git a/drivers/media/i2c/atomisp/ov2680.h b/drivers/media/i2c/atomisp/ov2680.h
index b2267b0..55d125e 100644
--- a/drivers/media/i2c/atomisp/ov2680.h
+++ b/drivers/media/i2c/atomisp/ov2680.h
@@ -84,7 +84,7 @@
 #define OV2680_FINE_INTG_TIME_MIN 0
 #define OV2680_FINE_INTG_TIME_MAX_MARGIN 0
 #define OV2680_COARSE_INTG_TIME_MIN 1
-#define OV2680_COARSE_INTG_TIME_MAX_MARGIN (0xffff - 6)
+#define OV2680_COARSE_INTG_TIME_MAX_MARGIN 6
 
 /*
  * OV2680 System control registers
diff --git a/drivers/media/i2c/atomisp/ov5648.c b/drivers/media/i2c/atomisp/ov5648.c
index 65092d3..4e97410 100644
--- a/drivers/media/i2c/atomisp/ov5648.c
+++ b/drivers/media/i2c/atomisp/ov5648.c
@@ -537,7 +537,7 @@ static long ov5648_s_exposure(struct v4l2_subdev *sd,
 	}
 
 	// EXPOSURE CONTROL DISABLED FOR INITIAL CHECKIN, TUNING DOESN'T WORK
-	return 0; // ov5648_set_exposure(sd, exp, gain, digitgain);
+	return ov5648_set_exposure(sd, exp, gain, digitgain);
 }
 
 static long ov5648_ioctl(struct v4l2_subdev *sd, unsigned int cmd, void *arg)
diff --git a/drivers/media/i2c/atomisp/ov5648.h b/drivers/media/i2c/atomisp/ov5648.h
index 15632d5..24eb543 100644
--- a/drivers/media/i2c/atomisp/ov5648.h
+++ b/drivers/media/i2c/atomisp/ov5648.h
@@ -81,7 +81,7 @@
 #define OV5648_FINE_INTG_TIME_MIN 0
 #define OV5648_FINE_INTG_TIME_MAX_MARGIN 0
 #define OV5648_COARSE_INTG_TIME_MIN 1
-#define OV5648_COARSE_INTG_TIME_MAX_MARGIN (0xffff - 6)
+#define OV5648_COARSE_INTG_TIME_MAX_MARGIN 6
 
 #define OV5648_BIN_FACTOR_MAX 4
 /*
-- 
2.1.0

